.travis.yml

```diff
@@ -0,0 +1,17 @@
+language: bash
+
+git:
+  depth: false
+
+before_install:
+  - sudo apt-get update
+  - sudo apt-get -y install shellcheck
+
+stages:
+  - code-check
+
+jobs:
+  include:
+    - stage: code-check
+      name: "bash shellcheck"
+      script: shellcheck */*.sh
```

Comment:
Anyone who fixes 1 line and sees 100s of warnings reported on code that has not changed will immediately start ignore this check and will never look back.

At the very least this check should run only on the files changed in the PR. Or even better: https://github.com/AtakamaLLC/lint-diffs (not tested)

Reply:
@marc-hb so you think this will only test with patches or the whole files?
I can refine this.